coderd/healthcheck/provisioner.go

```diff
@@ -96,19 +109,24 @@ func (r *ProvisionerDaemonsReport) Run(ctx context.Context, opts *ProvisionerDae
 			continue
 		}
 
-		r.ProvisionerDaemons = append(r.ProvisionerDaemons, db2sdk.ProvisionerDaemon(daemon))
+		it := ProvisionerDaemonsReportItem{
+			ProvisionerDaemon: db2sdk.ProvisionerDaemon(daemon),
+			Warnings:          make([]health.Message, 0),
+		}
 
 		// For release versions, just check MAJOR.MINOR and ignore patch.
 		if !semver.IsValid(daemon.Version) {
 			if r.Severity.Value() < health.SeverityError.Value() {
 				r.Severity = health.SeverityError
 			}
-			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Provisioner daemon %q reports invalid version %q", opts.CurrentVersion, daemon.Version))
+			r.Warnings = append(r.Warnings, health.Messagef(health.CodeUnknown, "Some provisioner daemons report invalid version information."))
```

Comment:
nit: maybe we should move all messages to a single file

Reply:
https://github.com/coder/coder/issues/11493